### PR TITLE
chore: improve the Indexer

### PR DESCRIPTION
**Pull request by muralov**
**Description**

Changes proposed in this pull request:

* Index with batches to prevent rate limiting
* Clean-up table after test run
* Add more logs

**Related issue(s)**
#199

### CODE REVIEW

File: doc_indexer/src/utils/models.py [hunk 1]
```diff
@@ -6,6 +6,8 @@
 from gen_ai_hub.proxy.langchain.openai import OpenAIEmbeddings
 from langchain_core.embeddings import Embeddings
 
+# TODO: re-use the model factory parent project
```
Comment (mfaizanse) [inline doc_indexer/src/utils/models.py:9]: Is it todo suppose to be completed as part of this PR?

File: doc_indexer/src/utils/settings.py [hunk 2]
```diff
@@ -5,6 +5,8 @@
 from decouple import Config, RepositoryEnv, config
 from dotenv import find_dotenv, load_dotenv
 
+# TODO: re-use the settings parent project
```
Comment (mfaizanse) [inline doc_indexer/src/utils/settings.py:8]: Is it todo suppose to be completed as part of this PR?

File: doc_indexer/src/utils/settings.py [hunk 3]
```diff
@@ -30,7 +32,9 @@ def is_running_pytest() -> bool:
 EMBEDDING_MODEL_DEPLOYMENT_ID = config("EMBEDDING_MODEL_DEPLOYMENT_ID")
 EMBEDDING_MODEL_NAME = config("EMBEDDING_MODEL_NAME")
 
-DOCS_PATH = config("DOCS_PATH", default=None)
+DOCS_PATH = config("DOCS_PATH", default="data/output")
+DOCS_TABLE_NAME = config("DOCS_TABLE_NAME", default="kyma_docs")
+CHUNKS_BATCH_SIZE = config("CHUNKS_BATCH_SIZE", cast=int, default=100)
```
Comment (mfaizanse) [inline doc_indexer/src/utils/settings.py:37]: Is `100` the best possible chunk size?

File: doc_indexer/src/utils/settings.py (hunk 2 above)
Comment (muralov) [reply doc_indexer/src/utils/settings.py:8]: it should be part of this [issue](https://github.com/kyma-project/kyma-companion/issues/243). Do you want me to add the issue link to the comment too?

File: doc_indexer/src/utils/models.py (hunk 1 above)
Comment (muralov) [reply doc_indexer/src/utils/models.py:9]: it should be part of this [issue](https://github.com/kyma-project/kyma-companion/issues/243). Do you want me to add the issue link to the comment too?

Comment (mfaizanse) [reply doc_indexer/src/utils/models.py]: Please remove the todo comment from the code then. Having it in the issue is enough.

File: doc_indexer/src/utils/settings.py (hunk 2 above)
Comment (mfaizanse) [reply doc_indexer/src/utils/settings.py]: Please remove the todo comment from the code then. Having it in the issue is enough.

**Review by mfaizanse — APPROVED** [review]

**Pull request merged**